datadog/data_source_datadog_synthetics_locations.go

```diff
@@ -0,0 +1,47 @@
+package datadog
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceDatadogSyntheticsLocations() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDatadogSyntheticsLocationsRead,
+
+		// Locations are a map of IDs to names
+		Schema: map[string]*schema.Schema{
+			"locations": {
+				Type:     schema.TypeMap,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceDatadogSyntheticsLocationsRead(d *schema.ResourceData, meta interface{}) error {
+
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClientV1 := providerConf.DatadogClientV1
+	authV1 := providerConf.AuthV1
+
+	syntheticsLocations, _, err := datadogClientV1.SyntheticsApi.ListLocations(authV1).Execute()
+
+	if err != nil {
+		return err
+	}
+
+	locationsMap := make(map[string]string)
+
+	for _, location := range syntheticsLocations.GetLocations() {
+		locationsMap[location.GetId()] = location.GetName()
+	}
+
+	if len(locationsMap) > 0 {
+		d.SetId("datadog-synthetics-location")
+		d.Set("locations", locationsMap)
+	} else {
+		d.Set("locations", locationsMap)
+	}
```

Comment:
What's this for ?

Reply:
We only set the ID if it's not empty, but we still set the map. I guess it should be outside the condition, let me fix it.